tutorials/23_MultiHop_QA_With_Agents.ipynb

```diff
@@ -0,0 +1,5680 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "mgQAP-4T7Fh6"
+      },
+      "source": [
+        "### 3) Ask a question!\n",
+        "\n",
+        "Now that you have an ExtractiveQAPipeline, go ahead and ask it a question about the presidents of the USA. \n",
+        "\n",
+        "Below is the question: \"Who was the 44th president of the USA?\"\n",
+        "\n",
+        "To start seeing the limitations of an ExtractiveQAPipeline on it's own, you may want to try a question like: \"Who is the wife of the 44th president of the USA?\".\n",
+        "\n",
+        "Notice hoe this is 2 questions in one. An extractive model will strugle to find the answer to this question unless the answer is phrased clearly in our documents. For example: \"The wife of the 44th president is Michelle Obama\"."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 930,
+          "referenced_widgets": [
+            "d31a483cf6e443a5891ca14672ffed3b",
+            "a0dcb1d860704bbdb54447dfd0cc92bd",
+            "5e85990cb7374ddc9f7107a72f812e5e",
+            "acf1cdc5c6484d8ab5f78ac217656b17",
+            "083cb8b3aa1240129a75e5c2a7eacbb6",
+            "7f4d17a4998f44b58f813b815d16f7f8",
+            "6ffba94cc23846209c4629cb10bce042",
+            "de52fc90bd3f4db0b540535246b33c4f",
+            "acd5844ab2d64071a1d59a1df7baafbd",
+            "744746530a4e401bb6e9e28d89e54eed",
+            "3bb3e922fc8c471a8c1aaaf5bbca2d92"
+          ]
+        },
+        "id": "HdS-satXbSsw",
+        "outputId": "751e30c8-50ae-48d3-8e34-df7023d1252a"
+      },
+      "outputs": [],
+      "source": [
+        "from haystack.utils import print_answers\n",
+        "\n",
+        "result = presidents_qa.run(\"Who was the 44th president of the USA\")\n",
+        "\n",
+        "print_answers(result, \"minimum\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bn83I_Dg8Q1f"
+      },
+      "source": [
+        "## Create an Agent with the `ExtractiveQAPipeline` as a `Tool`\n",
+        "### 1) To create the Agent, we will make use of an OpenAI model. So first, provide your OpenAI Key "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "kBkMr86vvl9h",
+        "outputId": "8366a06d-9064-485e-aa88-c82859c68265"
+      },
+      "outputs": [],
+      "source": [
+        "from getpass import getpass\n",
+        "\n",
+        "api_key_prompt = \"Enter OpenAI API key:\" \n",
+        "api_key = getpass(api_key_prompt)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "qajsCzjx9C6j"
+      },
+      "source": [
+        "### 2) Create the Agent \n",
```

Comment:
I'd explain here in 2-3 sentences how the agent uses the PromptNode and the LLM. Something along the lines of: 
- agents needs to determine next best course of action
- Under the hood the agent uses a LLM with a certain prompt to decide that
- It looks at the query we want to solve, the previous results it got from tools (all as plain texts) and then decided what to do next
- In case it decided to use another tool, it selects the tools from a list and prepares the input
- The LLM and a good prompt is key for a successful agent
- here we use OpenAI + zeroshot react prompt template

Reply:
I explain this quite a bit in overview so I'll create a condensed version of it here